### PR TITLE
Implement CER-71 Controllers enabling

### PR DESCRIPTION
**Pull request by andrew-certora**
Implements https://linear.app/euler-labs/issue/CER-71/controllers-enabling

### CODE REVIEW

**Review by kasperpawlowski — DISMISSED** [review]

**Review by kasperpawlowski — APPROVED** [review]

File: certora/specs/CER-11-Controllers/CER-71-Controllers-enabling.spec [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    address account;
+    address vault;
+    // Assume: EVC cannot be the caller of enableController.
+    // This assumption is proved in CER-51.
+    require e.msg.sender != currentContract;
+    // Caller is either an owner or operator of the account
+    require haveCommonOwner(account, e.msg.sender) || isAccountOperatorAuthorized(account, e.msg.sender);
+    // it is possible for this caller to succesfully enable a controller.
+    enableController@withrevert(e, account, vault);
+    satisfy !lastReverted;
```
Comment (nd-certora) [inline certora/specs/CER-11-Controllers/CER-71-Controllers-enabling.spec:44]: maybe better to make sure that both owner and operator can enable. so, two satisfy commands
```
    enableController(e, account, vault);
    satisfy haveCommonOwner(account, e.msg.sender);
    satisfy isAccountOperatorAuthorized(account, e.msg.sender);
```
but need the new `--independent_satisfy` flag

File: certora/conf/CER-11-Controllers/CER-71-Controllers-enabling.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "files": [
+    "certora/harness/EthereumVaultConnectorHarness.sol"
+  ],
+  "verify": "EthereumVaultConnectorHarness:certora/specs/CER-11-Controllers/CER-71-Controllers-enabling.spec",
+  "solc": "solc8.20",
+  "solc_optimize": "10000",
+  "solc_via_ir": true,
+  "msg": "CER-11-Controllers/CER-71-Controllers-enabling",
+  "optimistic_loop": true,
+  "optimistic_hashing": true
```
Comment (nd-certora) [inline certora/conf/CER-11-Controllers/CER-71-Controllers-enabling.conf:11]: rule_sanity basic

File: certora/specs/CER-11-Controllers/CER-71-Controllers-enabling.spec (hunk 1 above)
Comment (andrew-certora) [reply certora/specs/CER-11-Controllers/CER-71-Controllers-enabling.spec]: I rewrote it this way.

File: certora/conf/CER-11-Controllers/CER-71-Controllers-enabling.conf (hunk 2 above)
Comment (andrew-certora) [reply certora/conf/CER-11-Controllers/CER-71-Controllers-enabling.conf:13]: Good catch. I fixed this here.

**Review by nd-certora — APPROVED** [review]

**Pull request merged**